sql/mysql/tidb.go

```diff
@@ -0,0 +1,123 @@
+package mysql
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strings"
+
+	"ariga.io/atlas/sql/internal/sqlx"
+	"ariga.io/atlas/sql/schema"
+)
+
+type tinspect struct {
+	*inspect
+}
+
+func (i *tinspect) InspectSchema(ctx context.Context, name string, opts *schema.InspectOptions) (*schema.Schema, error) {
+	s, err := i.inspect.InspectSchema(ctx, name, opts)
+	if err != nil {
+		return nil, err
+	}
+	return i.patchSchema(ctx, s)
+}
+
+func (i *tinspect) InspectRealm(ctx context.Context, opts *schema.InspectRealmOption) (*schema.Realm, error) {
+	r, err := i.inspect.InspectRealm(ctx, opts)
+	if err != nil {
+		return nil, err
+	}
+	for _, s := range r.Schemas {
+		_, err := i.patchSchema(ctx, s)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return r, nil
+}
+
+func (i *tinspect) patchSchema(ctx context.Context, s *schema.Schema) (*schema.Schema, error) {
+	for _, t := range s.Tables {
+		if err := i.createStmt(ctx, t); err != nil {
+			return nil, err
+		}
+		if err := i.setCollation(t); err != nil {
+			return nil, err
+		}
+		if err := i.setFKs(s, t); err != nil {
+			return nil, err
+		}
+	}
+	return s, nil
+}
+
+// e.g CONSTRAINT "" FOREIGN KEY ("foo_id") REFERENCES "foo" ("id")
+var reFK = regexp.MustCompile("(?i)CONSTRAINT\\s+[\"`]*(\\w+)[\"`]*\\s+FOREIGN\\s+KEY\\s*\\(([,\"` \\w]+)\\)\\s+REFERENCES\\s+[\"`]*(\\w+)[\"`]*\\s*\\(([,\"` \\w]+)\\)")
```

Comment:
This is the regexp from SQLite? I think it can be simplified a lot because TiDB/MySQL "SHOW CREATE .." statement returns a formatted (and consistent) output.

Reply:
So use backtics to mark the regex string. Gets rid of all the escaped quotes

Reply:
simplified